benchmarks/sa.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python3                                                         
+
+import time, argparse
+import numpy as np
+import arkouda as ak
+
+TYPES = ('int64', 'float64', 'bool', 'str')
+
+def time_ak_sa( vsize, trials, dtype):
+    print(">>> arkouda suffix array")
+    cfg = ak.get_config()
+    Nv = vsize * cfg["numLocales"]
+    print("numLocales = {},  num of strings  = {:,}".format(cfg["numLocales"], Nv))
+#    v = ak.random_strings_uniform(90000000, 100000000, Nv)
+    v = ak.random_strings_uniform(1, 16, Nv)
+    c=ak.suffix_array(v)
+    print("size of suffix array={}".format(c.bytes.size))    
+#    print("All the random strings are as follows")
+    for k in range(vsize):
+       print("the {} th random tring ={}".format(k,v[k]))    
+       print("the {} th suffix array ={}".format(k,c[k]))    
+       print("")
+#    print(v)    
+    timings = []
+    for _ in range(trials):
+        start = time.time()
+        ak.suffix_array(v)
+        end = time.time()
+        timings.append(end - start)
+    tavg = sum(timings) / trials
+
+    print("Average time = {:.4f} sec".format(tavg))
+    if dtype == 'str':
+        offsets_transferred = 3 * c.offsets.size * c.offsets.itemsize
+        bytes_transferred = (c.offsets.size * c.offsets.itemsize) + (2 * c.bytes.size)
+        bytes_per_sec = (offsets_transferred + bytes_transferred) / tavg
+    else:
+        bytes_per_sec = (c.size * c.itemsize * 3) / tavg
+#    print("Average rate = {:.2f} GiB/sec".format(bytes_per_sec/2**30))
+
+def time_np_sa(Ni, Nv, trials, dtype, random):
+    print("to be done")
+
+def check_correctness(dtype, random):
+    print("to be done")
```

Comment:
Note that in order to run as part of CI testing this test should get added to the `BENCHMARKS` var in `benchmarks/run_benchmarks.py`

If we want this to run as part of nightly performance testing and get graphed, there's some other infrastructure that would be needed, but I don't think that's important for the initial PR.

Reply:
thanks for letting me know the CI testing problem.